### ISSUE TITLE
Allow indels in primer detection

### ISSUE THREAD

**Issue by mcroxen** [nextstrain/nextclade#510]
Thank you so much for your work on nextclade.

I was wondering if code could be added so that ```--input-pcr-primers primers.csv``` could also report on if there is a deletion or insertion in the priming region.

Thank you

**Comment by ivan-aksamentov** [comment]
If someone wants to implement, we discussed this with Matthew on Slack and here are some technical details:

Currently it only checks for substitutions. Deletions should be pretty easy to add, just need to loop over the deletions similar to how it loops over mutations:
https://github.com/nextstrain/nextclade/blob/15daff812fbaa8aff54ca8b7b846fe0cacc90850/packages/nextclade/src/analyze/getPcrPrimerChanges.cpp#L38-L44
and check if a deletion range and a primer range have an intersection (in terms of nuc positions).

Insertions might be a bit harder, they are stripped away early on, but should be doable too. Just need to pass all the necessary data into the function I linked above (or to make a new one).

**Comment by rneher** [comment]
yes, this would be a useful addition. We already have code to detect overlap of deletions with genes and codons, I believe. 

@mcroxen out of curiosity, what primer sets are you testing against? we have only a vague ideas of who uses this feature and the default primers in the app are possibly quite out-dated. So we'd be happy to find out how this us used.

**Comment by mcroxen** [comment]
@rneher we have our own in-house oligos that we like to keep an eye on, and thought it would be handy to be able to capture both substitutions and indels if they occur.
